flare/mgp/mapxb.py

```diff
@@ -445,79 +491,86 @@ def update_bounds(self, GP):
 
         lower_bound = self.bounds[0]
         min_dist = self.search_lower_bound(GP)
-        if min_dist < np.max(lower_bound): # change lower bound
-            warnings.warn('The minimal distance in training data is lower than \
-                    the current lower bound, will reset lower bound')
-
-        if self.auto_lower or (min_dist < np.max(lower_bound)):
-            lower_bound = np.max((min_dist - self.lower_bound_relax, 0))
+        # change lower bound only when there appears a smaller distance
+        if lower_bound is None or min_dist < np.max(lower_bound):
+            lower_bound = np.max((min_dist - self.lower_bound_relax, 0.0))
             rebuild_container = True
 
+            warnings.warn(
+                f"The minimal distance in training data is lower than "
+                f"the current lower bound, will reset lower bound to {lower_bound}"
+            )
+
         upper_bound = self.bounds[1]
-        if self.auto_upper:
-            upper_bound = Parameters.get_cutoff(self.kernel_name,
-                self.species, GP.hyps_mask)
-            rebuild_container = True
+        if self.auto_upper or upper_bound is None:
+            gp_cutoffs = Parameters.get_cutoff(
+                self.kernel_name, self.species, GP.hyps_mask
+            )
+            if upper_bound is None or np.any(gp_cutoffs > upper_bound):
+                upper_bound = gp_cutoffs
+                rebuild_container = True
 
         if rebuild_container:
             self.set_bounds(lower_bound, upper_bound)
             self.build_map_container()
 
-
     def build_map(self, GP):
 
         self.update_bounds(GP)
 
         if not self.load_grid:
             y_mean, y_var = self.GenGrid(GP)
         else:
-            if 'mgp_grids' not in os.listdir(self.load_grid):
-                raise FileNotFoundError("Please set 'load_grid' as the location of mgp_grids folder")
+            if "mgp_grids" not in os.listdir(self.load_grid):
```

Comment:
maybe consider adding gp's name or date info to the folder name, so that files in this folder would not be overwritten.

Reply:
It seems to me that using gp's name or date still can not prevent overwriting, if you just use gp of the same name, or map on the same day. In what situation do you want to keep all the grids files while have the possibility of being overwritten?